### PR TITLE
Enable metrics

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
Working on https://issues.redhat.com/browse/OCPCLOUD-784

This PR is adding support for reporting following prometheus metrics and also starting controller-runtime metrics server to make these metrics available for prometheus servers:

- `mapi_instance_create_failed`: Total count of "create" cloud api errors
- `mapi_instance_update_failed`: Total count of "update" cloud api errors
- `mapi_instance_delete_failed`: Total count of "delete" cloud api errors

Labels on these metrics (for Azure):
```golang
prometheus.Labels{
  "name": machine.Name,
  "namespace": machine.Namespace,
  "reason": error.Reason,
}
```

Depends on PR https://github.com/openshift/machine-api-operator/pull/609 introducing metrics support

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -27,3 +27,5 @@ require (
 	sigs.k8s.io/controller-tools v0.3.0
 	sigs.k8s.io/yaml v1.2.0
 )
+
+replace github.com/openshift/machine-api-operator => github.com/Danil-Grigorev/machine-api-operator v0.2.1-0.20200604073207-04068f243c6b
```
Comment (JoelSpeed) [inline go.mod:31]: I'd prefer this to be moved back to master openshift before merge
Hold until we can identify whether this is absolutely necessary
/hold

**Comment by JoelSpeed** [comment]
@Danil-Grigorev Seems there's a conflict, could you fix that then I'll give another review

File: go.mod [hunk 2]
```diff
@@ -23,11 +22,14 @@ require (
...
 	sigs.k8s.io/controller-runtime v0.6.0
 	sigs.k8s.io/controller-tools v0.3.0
 	sigs.k8s.io/yaml v1.2.0
 )
 
-replace sigs.k8s.io/cluster-api-provider-aws => github.com/openshift/cluster-api-provider-aws v0.2.1-0.20200612013036-1a97c62077cb
+replace (
+	sigs.k8s.io/cluster-api-provider-aws => github.com/openshift/cluster-api-provider-aws v0.2.1-0.20200618031251-e16dd65fdd85
+	sigs.k8s.io/cluster-api-provider-azure => github.com/openshift/cluster-api-provider-azure v0.1.0-alpha.3.0.20200618001858-af08a66b92de
+	sigs.k8s.io/cluster-api-provider-gcp => github.com/openshift/cluster-api-provider-gcp v0.0.1-0.20200617145800-5c21c4940146
```
Comment (enxebre) [inline go.mod:34]: why is this needed at all? gcp repo is not even a fork.

**Comment by enxebre** [comment]
why is this commit called "Add health checks"?

**Comment by enxebre** [comment]
this needs rebase.

**Comment by Danil-Grigorev** [comment]
@enxebre @JoelSpeed @alexander-demichev I'd like to merge those today. Please approve and LGTM
